security/pkg/pki/ca/ca.go

```diff
@@ -271,21 +292,24 @@ func NewPluggedCertIstioCAOptions(fileBundle SigningCAFileBundle,
 }
 
 // BuildSecret returns a secret struct, contents of which are filled with parameters passed in.
+// Adds the "istio-generated" key.
 func BuildSecret(scrtName, namespace string, certChain, privateKey, rootCert, caCert, caPrivateKey []byte, secretType v1.SecretType) *v1.Secret {
-	return &v1.Secret{
+	secret := &v1.Secret{
```

Comment:
nit: this doesn't add anything new, so we can likely just leave it as `return &v1.Secret{` since `secret` is returned immediately after anyway